### ISSUE TITLE
Issues with import

### ISSUE THREAD

**Issue by classbproject** [partkeepr/PartKeepr#855]
I've been trying to test an import with a few test parts. The headers of my CSV look like this: 

![2017-05-20_210745](https://cloud.githubusercontent.com/assets/19990466/26277006/15d5e7f6-3da1-11e7-844d-2e516da335c7.png)

I've selected the data fields for Distributors under System Preferences but they won't show up in the Import fields. 
![2017-05-20_211520](https://cloud.githubusercontent.com/assets/19990466/26277060/22a20374-3da2-11e7-8e74-2d88baba1be9.png)

![2017-05-20_211556](https://cloud.githubusercontent.com/assets/19990466/26277061/22a2f13a-3da2-11e7-8b23-8eaa72c0a19c.png)

Maybe I do not understand how to work this. Is there a guide that can help me? Using version 1.2.0.

**Comment by felicitus** [comment]
1) The system preferences for parts have nothing to do with the import, it just defines which fields are required when entering new part data

2) There seems to be a bug that m:n relations aren't being displayed, I'll have a look into it

**Comment by felicitus** [comment]
I found and fixed the bug; you can check it out in the latest GIT version or wait until the next version is being released.

**Comment by classbproject** [comment]
I tried this and I'm still not able to do it. I cannot see where I can enter distributor names, their part numbers, manufacturer names and currency. I'm not able to map these fields to the fields that show on the left since I cannot find them in the list. The import does not list an error now but neither does the part show.

**Comment by felicitus** [comment]
Have you properly updated from GIT? The demo site shows the associations for distributor's name, currency and orderNumber: 

![image](https://cloud.githubusercontent.com/assets/162974/26554304/baa5723e-448f-11e7-8944-3c83baf3f8ac.png)

I know that configuring the importer correctly is complex, but I currently don't have the spare time to prepare a documentation for that.

**Comment by classbproject** [comment]
I did the whole setup again on a different server and it doesn't show.
<img width="396" alt="screen shot 2017-05-29 at 20 28 50" src="https://cloud.githubusercontent.com/assets/19990466/26554404/82b3bee4-44ad-11e7-93cb-b38783d65d0b.png">

**Comment by felicitus** [comment]
Which version did you install?

**Comment by classbproject** [comment]
<img width="723" alt="screen shot 2017-05-29 at 20 30 58" src="https://cloud.githubusercontent.com/assets/19990466/26554452/c421e400-44ad-11e7-899e-52fd56e1e173.png">

**Comment by felicitus** [comment]
Hmm, the git revision looks OK to me. Can you make a screenshot when you scroll down the field list in the importer?

**Comment by classbproject** [comment]
<img width="376" alt="screen shot 2017-05-29 at 20 33 50" src="https://cloud.githubusercontent.com/assets/19990466/26554539/3d9dc3f8-44ae-11e7-8de4-81b89ba9c457.png">

**Comment by felicitus** [comment]
This is very odd - can you double-check if the field mappings appear when you are using the demo system?

**Comment by classbproject** [comment]
Yes they do but only after I opened it in an incognito window. I guess its cached somewhere. I then opened my installation in an incognito window but no joy :(

**Comment by felicitus** [comment]
Well to be honest, I've seen such a behavior in the past, where some associations disappear without any clear reason. I thought at first that in your case the GIT version was old or that this might be a browser issue, but I guess the issue is back.

Note for myself: Seems that I need to refactor storage of m:n associations somehow, as it seems that ExtJS is not reliable in that regards.

I'm sorry that I can't provide an immediate fix :( . If you have ideas on how to reproduce or to narrow down the issue please let me know.

**Comment by classbproject** [comment]
:( Let me try to reinstall this on a separate server again. The last time I installed it the installation took a while but this time when I installed from git I just breezed through it. Not sure if there was something that I did there which may be causing this. Once I get this working and manage to successfully import a few part numbers, I can help you write the documentation for that.

**Comment by jaimelaborda** [comment]
Hi,
Regarding this topic, I'm having problems to do some imports. Particularly I'm having some problems when trying to import parts with a Distributor or Manufacturer. Maybe I'm doing something wrong but I cannot manage to properly import Manufacturer or/and Distributor.

This is the .CSV file I'm trying to import:
```

Cust Part Number,Type,Description,Manufacturer 1,Manufacturer Part Number 1,Manufacturer 2,Manufacturer Part Number 2,Footprint,Cod. RS,Cod. Digikey,Cod. Farnell
URISEC0001,CAPACITOR,Cap Ceramic 1uF 10V X5R 10% 0603,KEMET,C0603C105K8PACTU,,,,461-2708 ,,
URISEC0002,CAPACITOR,Cap Ceramic 100nF 16V X7R 10% 0603,KEMET,C0603C104K4RACTU,,,,264-4630,,
```
There are two components. 

This is the steps I have follow (in demo.partkeepr.org):
1. Open import dialog
2. Upload the CSV file: Parts are loaded correctly no issue there.
3. I do select **partPartKeeprPartBundleEntityPartManufacturers** and then enter partNumber as "Copy content from:" (image attached)
![image](https://user-images.githubusercontent.com/12583019/80522958-a6335f00-898d-11ea-8072-4dd32d045318.png)

4. Select manufacturer "directory" and then the option to "Match import data with existing data using:". Then I select the manufacturer name match field with the Import Field in my CSV.
(image attached)
![image](https://user-images.githubusercontent.com/12583019/80522758-5eacd300-898d-11ea-8bb4-6f0f2e96b9aa.png)

These produce no result neither in the preview section or importing.

May I doing something wrong? There is some documentation to import parts¿? I have not found it.

Thank you very much.

Best regards,
Jaime.
